spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/GraphGeneratorVisitor.java

```diff
@@ -0,0 +1,651 @@
+						currentContext().addOtherExit(n.id);
+					} else {
+						currentContext().addDanglingNodes(false, n.id);
+					}
+				}
+			}
+		} else {
+			// Check if it is a transition to something labeled earlier in this flow
+			Node existingNode = currentContext().getNodeLabeled(transition.getTargetLabel());
+			if (existingNode != null) {
+				addLink(new Link(currentTaskAppId, existingNode.id, transition.getStatusToCheckInDSLForm()));
+			}
+			else {
+				// Record a new transition attempted
+				currentContext().addTransitionTarget(currentTaskAppId, transition.getStatusToCheckInDSLForm(),
+						transition.getTargetLabel());
+			}
+		}
+	}
+
+	public GraphGeneratorVisitor.Context currentContext() {
+		return contexts.peek();
+	}
+
+	public GraphGeneratorVisitor.Context parentContext() {
+		if (contexts.size() < 2) {
+			return null;
+		} else {
+			return contexts.get(contexts.size() - 2);
+		}
+	}
+	
+	public Sequence currentSequence() {
+		return sequences.get(currentSequence);
+	}
+
+	@Override
+	public void visit(TaskApp taskApp) {
+		String nextId = nextId();
+		currentTaskAppId = nextId;
+		Node node = new Node(nextId, taskApp.getName());
+		addNode(node);
+		if (taskApp.hasLabel()) {
+			node.setLabel(taskApp.getLabelString());
+			currentContext().addNodeWithLabel(taskApp.getLabelString(),node);
+		}
+		if (currentContext().isFlow) {
+			// Are there any outstanding transitions that need to be connected
+			// to this?
+			if (taskApp.hasLabel()) {
+				for (Iterator<Context.TransitionTarget> iterator = currentContext().getTransitionTargets()
+						.iterator(); iterator.hasNext();) {
+					Context.TransitionTarget tt = iterator.next();
+					if (tt.label.equals(taskApp.getLabelString())) {
+						// Target found!
+						addLink(new Link(tt.nodeId, nextId, tt.onState));
+						iterator.remove();
+					}
+				}
+			}
+			List<String> danglingNodes = currentContext().getDanglingNodes();
+			if (danglingNodes.size() == 0) {
+				// first entry
+				addLink(new Link(currentContext().startNodeId, nextId));
+				currentContext().addDanglingNodes(true, nextId);
+			} else {
+				for (int i = 0; i < danglingNodes.size(); i++) {
+					addLink(new Link(danglingNodes.get(i), nextId));
+				}
+				currentContext().addDanglingNodes(true, nextId);
+			}
+			currentContext().clearDanglingNodes();
+			currentContext().addDanglingNodes(true, nextId);
+		} else if (currentContext().isSplit) {
+			// Any flow containing this split can transition to this split if it
+			// is labeled. This code
+			// isn't complete as it will only link it to the first one in the
+			// split, when it needs to
+			// link to all of them. Or we create a sync node and link to that.
+			if (currentContext().containingNode.hasLabel() && (parentContext() != null && parentContext().isFlow)) {
+				// A surrounding flow can target a split with a transition
+				// target
+//				for (Iterator<Context.TransitionTarget> iterator = parentContext().getTransitionTargets()
```

Comment:
The graph generator visitor is a work in progress to verify the basic visitor support. There are known limitations in the generator that will be revisited when Flo starts using it, this code will be activated in some form at that time. I'd like to keep it for now.